delivery_service.rb

```diff
@@ -0,0 +1,36 @@
+class DeliveryService
+
+  attr_accessor :delivery_confirmed, :distance , :weight 
+
+  def initialize(weight, distance)    
+    @weight = weight
+    @distance = distance
+    @delivery_confirmed = false 
+    @park = Array.new
+    @park << Car.new(rand(9), 'xxx' , true) << Car.new(rand(9), 'xxx1' , false)  
+    @park << Bike.new(rand(5), true) << @park << Bike.new(rand(5), false)     
+  end 
+
+  def choose_transport_type
+    if @weight <= Bike::BIKE_MAX_WEIGHT && @distance <= Bike::BIKE_MAX_DISTANCE
```

Comment:
U transport should provide logic of availability. U could define basic `available?` method in `Transport` class and redefine it in `Bike` class for adding additional check

Reply:
So, It could be like
```ruby
@park.find(&:available?)
```
or with some arguments
```ruby
@part.find { |vehicle| vehicle.available?(*args) }
```

Reply:
Done